pageserver/src/tenant/timeline.rs

```diff
@@ -1903,6 +1903,7 @@ impl Timeline {
             .set((calculated_size, metrics_guard.calculation_result_saved()))
             .ok()
             .expect("only this task sets it");
+        self.current_logical_size.initialized.notify_one();
```

Comment:
Notify doesn't feel like the right concurrency primitive for this.
It's more like a OnceCell that others can wait-for-to-be-set.

Reply:
I've defensively converted both the Notify instances to semaphores, so that they aren't vulnerable to misbehaving if multiple callers try to await them.  It's not super elegant, but we can wrap it up into a "Kick" type condition variable-alike later if we need to.

911b1c6f21bf696024c8e27edd6c5746ca191dbb